pkg/compactor/compactor.go

```diff
@@ -218,11 +221,24 @@ type Compactor struct {
 	compactionRunInterval          prometheus.Gauge
 	blocksMarkedForDeletion        prometheus.Counter
 	garbageCollectedBlocks         prometheus.Counter
+	blocksMarkedForNoCompact       prometheus.Counter
 
 	// TSDB syncer metrics
 	syncerMetrics *syncerMetrics
 }
 
+type noOpsCounter struct {
```

Comment:
based on sync offline, I have created this noOpsCounter so that there's no metric when feature is disabled. I don't think this is elegant and I'm looking at some examples. Would appreciate some advice.

Reply:
@bboreham @pracucci @pstibrany it would be nice to have your thoughts regarding this as well - should we not emit the metric at all when "skip bad block" feature is disabled.

Reply:
You can just pass `prometheus.NewCounter(prometheus.CounterOpts{})` - if it isn't registered it doesn't go anywhere.